### ISSUE TITLE
Test Windows 10 compatibility

### ISSUE THREAD

**Issue by Nadahar** [UniversalMediaServer/UniversalMediaServer#640]
I doubt there are much changes needed for Windows 10, but if somebody has been crazy enough to upgrade, UMS should be tested. 

I've picked up one place where a change is needed, ```PmsConfiguration.isAdmin()``` (the behaviour is probably the same as for Windows 7 & 8, but it needs testing). There might be other places where it matters as well.

**Comment by SubJunk** [comment]
@Nadahar good point. I think @Optimusprimeums uses it and said it works well with UMS, is that right?

**Comment by valib** [comment]
@SubJunk I am also testing UMS on WIN 10 and it works without any problems.

**Comment by Nadahar** [comment]
@valib You should check out ```PmsConfiguration.isAdmin()``` then and make sure that the admin permissions check works also for Windows 10:
```java
	public boolean isAdmin() {
		if (
			"Windows 8".equals(System.getProperty("os.name")) ||
			"Windows 7".equals(System.getProperty("os.name")) ||
			"Windows Vista".equals(System.getProperty("os.name"))
		) {
			try {
				String command = "reg query \"HKU\\S-1-5-19\"";
				Process p = Runtime.getRuntime().exec(command);
				p.waitFor();
				int exitValue = p.exitValue();

				if (0 == exitValue) {
					return true;
				}

				return false;
			} catch (IOException | InterruptedException e) {
				LOGGER.error("Something prevented UMS from checking Windows permissions", e);
			}
		}

		return true;
```

**Comment by Nadahar** [comment]
@valib If you make a commit regarding that, you could fix the exception logging as well, as it is not it doesn't print the exception. Change:
```java
LOGGER.error("Something prevented UMS from checking Windows permissions", e);
```
to:
```java
LOGGER.error("Something prevented UMS from checking Windows permissions: {}", e);
```

**Comment by Optimusprimeums** [comment]
Windows 10 works fine for me, been running UMS on it since the the inside preview

**Comment by Nadahar** [comment]
Thanks @Optimusprimeums, I'll probably pick it up if nobody else does.

@valib @Optimusprimeums I'm not surprised that it works in general, but things like install service and any other "special Windows function" needs to be tested specifically. Have any of you tried to install UMS as a service in Windows 10?

**Comment by valib** [comment]
@Nadahar will try it when will be at home

**Comment by Nadahar** [comment]
@valib Did you ever get around to test this?

**Comment by Optimusprimeums** [comment]
@Nadahar Install and Uninstall as a Service works fine for me on x64bit Windows 10. Runs fine as well

**Comment by Nadahar** [comment]
@Optimusprimeums I will work fine as long as your user has admin permissions, but as it currently stands there's no way the admin check will work for Windows 10. That means that you're probably allowed to install as a service even if you're not an admin, and then that will fail.

**Comment by Optimusprimeums** [comment]
@Nadahar True, we need to add a check for that or the ability to call for admin rights when it does the install uninstall

**Comment by Nadahar** [comment]
@Opti I briefly looked at adding the ability to ask for admin rights when needed, but it seems it's close to impossible under java. For that to work a manifest must be embedded in the exe file, and since java doesn't compile to .exe you have to use some kind of a wrapper. Even if you wrap it in an .exe and manages to get the manifest right, you have to restart the application to elevate permissions, and there's no way to demote again without another restart. The best option would probably be to create a real executable in some other language that does the service installation/uninstallation and let that application ask for elevated permissions. 

It's much easier to just disable the button if UMS is not run with elevated permissions that to arrange all that, that's for sure, even though the solution would have been better.

**Comment by Optimusprimeums** [comment]
@Nadahar Very true, I think just disable it like you've said as i think doing the other stuff is too complex for a feature that might not be used by many. I've looked at similar ways to do it a few years ago but it's was too hard to implement as you've said above.

**Comment by Nadahar** [comment]
This is hopefully addressed with #670, so I'll close this one.
